runtime/src/message_processor.rs

```diff
@@ -64,6 +64,7 @@ impl MessageProcessor {
         blockhash: Hash,
         lamports_per_signature: u64,
         current_accounts_data_len: u64,
+        accumulated_consumed_units: &mut Option<&mut u64>,
```

Comment:
```suggestion
        accumulated_consumed_units: Option<&mut u64>,
```
Should suffice, though I don't see where the `Option` is adding anything?

Reply:
I left a reply  to your comment there, but I don't see `&mut None` being of much more value than
```
let mut _ignored: u64 = 0;
...
    &mut _ignored,
```

The option is never overwritten inside the call, is it?

Reply:
Well, `as_deref_mut()` didn't go too far either. I'll read Rust Book again, but for now, I removed `Option` in this[ commit](60b05c2b36207927131dcce124d6c393cf801b69) .